geomstats/_backend/tensorflow/__init__.py

```diff
@@ -532,6 +532,39 @@ def cast(x, dtype):
     return tf.cast(x, dtype)
 
 
+def broadcast_arrays(x, y, **kwargs):
+    tensors = [x, y]
+    shapes = [t.get_shape().as_list() for t in tensors]
+    max_rank = max(len(s) for s in shapes)
+
+    for index, value in enumerate(shapes):
+        shape = value
+        if len(shape) == max_rank:
+            continue
+
+        tensor = tensors[index]
+        for _ in range(max_rank - len(shape)):
+            shape.insert(0, 1)
+            tensor = tf.expand_dims(tensor, axis=0)
+        tensors[index] = tensor
+
+    from collections import Counter
+    broadcast_shape = []
+    for index in range(max_rank):
+        dimensions = [s[index] for s in shapes]
+        repeats = Counter(dimensions)
+        if len(repeats) > 2 or (len(repeats) == 2 and
+                                1 not in list(repeats.keys())):
+            raise Exception("Broadcasting not possible")
```

Comment:
Maybe give the reason why broadcasting is not possible, in the spirit of np's error message:

```ValueError: operands could not be broadcast together with shapes (2,3) (4,3)```